contrib/babelfishpg_tds/src/backend/tds/tdslogin.c

```diff
@@ -1989,41 +1989,45 @@ TdsProcessLogin(Port *port, bool loadedSsl)
 	TdsErrorContext->phase = 0;
```

Comment:
This is a really really large PR. We shouldn't merge this PR as a single commit as this is handling several new features together. Each feature requires its own explanation in the commit message:

Let's split the PR as per following:

1. JDBC test framework changes for jTDS driver testing
2. Prelogin changes
3. SP_cursorprepare
4. SP_cursorfetch with fetchinfo
5. Datatype changes (Individual datatypes should be addressed independently if the fixes are not related)

Each PR should be backed by respective testcases instead of putting all the test case in a single PR.